rcl/test/rcl/test_events.cpp

```diff
@@ -0,0 +1,546 @@
+  rcl_event_t publisher_event;
+  rcl_subscription_t subscription;
+  rcl_event_t subscription_event;
+  bool is_unsupported, is_opensplice, is_fastrtps;
+  const char * topic = "rcl_test_publisher_subscription_events";
+  const rosidl_message_type_support_t * ts;
+};
+
+rcl_ret_t
+wait_for_msgs_and_events(
+  rcl_subscription_t * subscription,
+  rcl_event_t * subscription_event,
+  rcl_event_t * publisher_event,
+  rcl_context_t * context,
+  int64_t period_ms,
+  bool & msg_ready,
+  bool & subscription_event_ready,
+  bool & publisher_event_ready)
+{
+  msg_ready = false;
+  subscription_event_ready = false;
+  publisher_event_ready = false;
+
+  int num_subscriptions = (nullptr == subscription ? 0 : 1);
+  int num_events = (nullptr == subscription_event ? 0 : 1) + (nullptr == publisher_event ? 0 : 1);
+
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rcl_ret_t ret = rcl_wait_set_init(&wait_set, num_subscriptions, 0, 0, 0, 0, num_events,
+    context, rcl_get_default_allocator());
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    rcl_ret_t ret = rcl_wait_set_fini(&wait_set);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  });
+
+  ret = rcl_wait_set_clear(&wait_set);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  if (nullptr != subscription) {
+    ret = rcl_wait_set_add_subscription(&wait_set, subscription, NULL);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+  if (nullptr != subscription_event) {
+    ret = rcl_wait_set_add_event(&wait_set, subscription_event, NULL);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+  if (nullptr != publisher_event) {
+    ret = rcl_wait_set_add_event(&wait_set, publisher_event, NULL);
+    EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+
+  ret = rcl_wait(&wait_set, RCL_MS_TO_NS(period_ms));
+  if (ret == RCL_RET_TIMEOUT) {
+    return ret;
+  }
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  for (size_t i = 0; i < wait_set.size_of_subscriptions; ++i) {
+    if (wait_set.subscriptions[i] && wait_set.subscriptions[i] == subscription) {
+      msg_ready = true;
+    }
+  }
+  for (size_t i = 0; i < wait_set.size_of_events; ++i) {
+    if (nullptr != wait_set.events[i]) {
+      if (wait_set.events[i] == subscription_event) {
+        subscription_event_ready = true;
+      } else if (wait_set.events[i] == publisher_event) {
+        publisher_event_ready = true;
+      }
+    }
+  }
+
+  return RCL_RET_OK;
+}
+
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_unsupported_lifespan) {
+  if (is_unsupported) {
+    rmw_time_t deadline {0, 0};
+    rmw_time_t lifespan {1, 0};
+    rmw_time_t lease_duration {1, 0};
+    rmw_qos_liveliness_policy_t liveliness_policy = RMW_QOS_POLICY_LIVELINESS_AUTOMATIC;
+    EXPECT_EQ(RMW_RET_ERROR,
+      setup_subscriber(deadline, lifespan, lease_duration,
```

Comment:
Doesn't this return an `rcl_ret_t`? Shouldn't you compare it to an `RCL_RET_*` define?

Reply:
Yes this should be RCL_RET_ERROR

Reply:
Will change to `RCL_RET_ERROR`.